xbmc/FileItem.cpp

```diff
@@ -1093,7 +1093,10 @@ bool CFileItem::IsNFO() const
 
 bool CFileItem::IsDiscImage() const
 {
-  return URIUtils::HasExtension(m_strPath, ".img|.iso|.nrg|.udf");
+  if (URIUtils::HasExtension(m_strPath))
```

Comment:
dynpath should always take precedence over strPath (as you can see in all other implementations). When you call `GetDynPath` on `CFileItem` it will return the dynpath if not empty and then fall back to path. If you write this the other way around, does it still work?

Reply:
I'd say to use just:

`return URIUtils::HasExtension(GetDynPath(), ".img|.iso|.nrg|.udf");`

Reply:
Makes sense.  I kept as close to the original code as possible because for some routes, `m_strPath` worked correctly and I was wary of introducing any regression.